lapis2/src/main/kotlin/org/genspectrum/lapis/model/AggregatedModel.kt

```diff
@@ -34,6 +35,25 @@ class AggregatedModel(private val siloClient: SiloClient) {
         if (key == "pangoLineage") {
             return PangoLineageEquals(value)
         }
+        if (key == "nucleotideMutations") {
+            return generateNucleotideFilter(value)
+        }
         return StringEquals(key, value)
     }
+
+    private fun generateNucleotideFilter(userInput: String): SiloFilterExpression {
+        val mutations = userInput.split(",")
+
+        return And(
+            mutations.map { mutationExpression ->
+                val match = Regex("(?<symbolFrom>^[A-Za-z]{0,})?(?<position>\\d{1,})(?<symbolTo>[A-Za-z]\$)")
```

Comment:
A nucleotide mutation expression can also contain deletions represented by a `-`. See https://github.com/GenSpectrum/LAPIS/blob/e00d42f27ac4ae7215189c15c89b343534a860d9/server/src/main/antlr/ch/ethz/lapis/api/parser/VariantQuery.g4#L41

Reply:
`.` is syntactic sugar for _not mutated_. Example: If the reference genome has an `A` at position 100, then `100.` is equivalent to `100A`.

I don't know whether SILO supports the syntax and I don't know whether it should. As it is purely for convenience, we might want to only support it in LAPIS. @Taepper, what do you think?

Reply:
SILO has the filters `NucleotideEquals` and `HasNucleotideMutation`. We implemented the mapping to `NucleotideEquals`. `100.` corresponds to `Not(HasNucleotideMutation)`, doesn't it?
But SILO also already supports `100.` by mapping it to `100<looked up from reference genome>`.

https://github.com/GenSpectrum/LAPIS-SILO/blob/main/src/silo/query_engine/query_engine.cpp#L135-L150

Reply:
I think they are different when it comes to unknowns. By default, `HasNucleotideMutation` should only be true if a sequence has a confirmed mutation. If a sequence has an `N`, it will be evaluated as false. `Not(HasNucleotideMutation)` is then true. `.`, on the other hand, is true if a sequence has definitely the reference base.

If SILO already supports `.`, we should use it.